module2/src/com/epam/arraysofarrays/MatrixWorkerOrderN.java

```diff
@@ -0,0 +1,69 @@
+package com.epam.arraysofarrays;
+
+public class MatrixWorkerOrderN {
+    //task 4
+    public static  int[][] createSquareMatrixOrderPositiveN (int positiveN) {
+        int [][] matrix = new int[positiveN][positiveN];
+        for( int i = 0; i < positiveN; i++) {
+            for( int j = 0; j < positiveN; j++) {
+                if (i % 2 ==0) {
+                    matrix[i][j] = j + 1;
+                } else {
+                    matrix[i][j] = positiveN - j;
+                }
+            }
+        }
+        return matrix;
+    }
+    // task 5
+    public static int[][] createMatrixForTaskFive(int orderN) {
+        int [][] matrix = new int[orderN][orderN];
+        for( int i = 0; i < orderN; i++) {
+            for( int j = 0; j < orderN; j++) {
+                if ( i + j < orderN) {
+                    matrix[i][j] = i + 1;
+                } else {
+                    matrix[i][j] = 0;
+                }
+            }
+        }
+        return matrix;
+    }
+    // task 6
+    public static int[][] createMatrixForTaskSix(int orderN) {
+        int [][] matrix = new int[orderN][orderN];
+        for( int i = 0; i < orderN / 2; i++) {
+            for( int j = 0; j < orderN; j++) {
+                if(((i > j) || (i >= orderN - j ))) {
+                    matrix[i][j] = 0;
+                } else {
+                    matrix[i][j] = 1;
+                }
+            }
+        }
+        for(int i = (orderN / 2); i < orderN; i++) {
+            for( int j = 0; j < orderN; j++) {
+                if(matrix[orderN - i - 1][j] == 1 ) {
+                    matrix[i][j] = 1;
+                } else {
+                    matrix[i][j] = 0;
+                }
+            }
+        }
```

Comment:
Basically you can do everything in 1 cycle
```suggestion
            final int rowBeforeMiddle = (matrix.length / 2) - 1;
            int numberOfZeros = 0;
            for (int i = 0; i < matrix.length; i++) {
                for (int j = numberOfZeros; j < (matrix[i].length - numberOfZeros); j++) {
                    matrix[i][j] = 1;
                }
                if (i < rowBeforeMiddle) {
                    numberOfZeros++;
                } else if (i > rowBeforeMiddle) {
                    numberOfZeros--;
                }
            }
```